src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CancellationTokenParametersMustComeLast.cs

```diff
@@ -99,6 +100,20 @@ public override void Initialize(AnalysisContext context)
                         last--;
                     }
 
+                    // Ignore parameters that hav any of these attributes.
+                    // C# reserved attributes: https://docs.microsoft.com/dotnet/csharp/language-reference/attributes/caller-information
+                    var callerInformationAttributes = ImmutableArray.Create<INamedTypeSymbol>(
+                        compilationContext.Compilation.GetTypeByMetadaName(typeof(CallerFilePathAttribute).FullName),
+                        compilationContext.Compilation.GetTypeByMetadaName(typeof(CallerLineNumberAttribute).FullName),
+                        compilationContext.Compilation.GetTypeByMetadaName(typeof(CallerMemberNameAttribute).FullName));
+
+                    while (last >= 0
+                        && methodSymbol.Parameters[last].IsOptional
```

Comment:
This is just to avoid the loop in `HasCallerInformationAttribute` if the parameter isn't optional.